### PR TITLE
Avoid some intermediate `List`s

### PR DESCRIPTION
**Pull request by lolgab**

### CODE REVIEW

File: compiler/src/dotty/tools/MainGenericRunner.scala [hunk 1]
```diff
@@ -148,7 +148,7 @@ object MainGenericRunner {
     case (o @ javaOption(striped)) :: tail =>
       processArgs(tail, settings.withJavaArgs(striped).withScalaArgs(o))
     case (o @ scalaOption(_*)) :: tail =>
-      val remainingArgs = (CommandLineParser.expandArg(o) ++ tail).toList
+      val remainingArgs = (CommandLineParser.expandArg(o) ::: tail).toList
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/MainGenericRunner.scala:151]:
```suggestion
      val remainingArgs = CommandLineParser.expandArg(o) ::: tail
```

File: compiler/src/dotty/tools/dotc/core/Definitions.scala [hunk 2]
```diff
@@ -2132,7 +2132,7 @@ class Definitions {
     if (!isInitialized) {
       // force initialization of every symbol that is synthesized or hijacked by the compiler
       val forced =
-        syntheticCoreClasses ++ syntheticCoreMethods ++ ScalaValueClasses() :+ JavaEnumClass
+        syntheticCoreClasses ::: syntheticCoreMethods ++ ScalaValueClasses() :+ JavaEnumClass
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/core/Definitions.scala:2135]: This one is a bit strange. Maybe we could force them in a simpler way.

**Review by nicolasstucki — COMMENTED** [review]
Otherwise LGTM

**Comment by dwijnand** [comment]
I don't think we should bother with this.  List.++ calls directly calls concat, which is overridden in List to match the suffix (the argument) to List and then call :::.  So we're removing that overhead, which is constant, so not worth it.

**Comment by nicolasstucki** [comment]
I don't see an override for `concat` in [`List`](https://github.com/scala/scala/blob/2.13.x/src/library/scala/collection/immutable/List.scala) or its parents.

**Comment by plokhotnyuk** [comment]
How about replacing usage of `List` by `ArraySeq` (or just `Array` in method implementations) to reduce allocations during building and concatenations and for faster access during reading?

**Comment by lolgab** [comment]
I dropped the change from `++` to `:::` and left the only changes that seem to make sense now.

**Review by dwijnand — APPROVED** [review]
Looks fine now

**Pull request merged**